test/integ/test_connection.py

```diff
@@ -1192,3 +1193,47 @@ def check_packages(message: str, expected_packages: list[str]) -> bool:
     ) as conn, capture_sf_telemetry.patch_connection(conn, False) as telemetry_test:
         conn._log_telemetry_imported_packages()
         assert len(telemetry_test.records) == 0
+
+
+@pytest.mark.skipolddriver
+@pytest.mark.parametrize(
+    "mode",
+    ("file", "env"),
+)
+def test_connection_name_loading(monkeypatch, db_parameters, tmp_path, mode):
+    import tomlkit
+
+    doc = tomlkit.document()
+    default_con = tomlkit.table()
+    tmp_config_file: None | pathlib.Path = None
+    try:
+        # If anything unexpected fails here, don't want to expose password
+        for k, v in db_parameters.items():
+            default_con[k] = v
+        with monkeypatch.context() as m:
+            if mode == "env":
+                doc["default"] = default_con
+                m.setenv("SF_CONNECTIONS", tomlkit.dumps(doc))
+            else:
+                doc["connections"] = tomlkit.table()
+                doc["connections"]["default"] = default_con
+                tmp_config_file = tmp_path / "config.toml"
+                tmp_config_file.write_text(tomlkit.dumps(doc))
+            with snowflake.connector.connect(
+                connection_name="default",
+                config_file_path=tmp_config_file,
+            ) as conn:
+                with conn.cursor() as cur:
+                    assert cur.execute("select 1;").fetchall() == [
+                        (1,),
+                    ]
+    except Exception:
```

Comment:
Why not just let exceptions percolate up?

Reply:
Because I'm actually operating on `db_parameters` which has our connection options in it (including password and other sensitive information). pytest has a very useful feature where if an Exception is raised all the way up it'll print local variables from the stack. This is my way of guaranteeing that we'll never expose sensitive information.
This is my first time Catching and Swallowing all exceptions in a test like this though, so I'm open to suggestions on doing this more elegantly.

Reply:
You're reasoning is sound; I'd just ask that if you keep it this way, please add a comment for your future self because it *looks* odd.  A comment will forestall any future headscratching.

That said, another approach would be to sanitize the exception if you knew where sensitive information could possibly reside.

As a principle, I see `except Exception` mostly on par with bare-`except`.  Generally these should be avoided unless you're catching all exceptions, doing some extra work, and then re-raising the original exception or chaining the original exception to a new one.  The classic example of course is aborting a transaction on an exception.

Thanks for the explanation.  I think whether you keep it or sanitize-and-reraise, please just add a comment explaining your rationale.